cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterUserCatalogController.java

```diff
@@ -0,0 +1,112 @@
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import cwms.cda.api.errors.CdaError;
+import cwms.cda.data.dao.watersupply.WaterContractDao;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.watersupply.WaterUser;
+import cwms.cda.data.dto.watersupply.WaterUserContract;
+import cwms.cda.formatters.ContentType;
+import cwms.cda.formatters.Formats;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+
+public final class WaterUserCatalogController extends WaterSupplyControllerBase implements Handler {
+    private static final Logger LOGGER = Logger.getLogger(WaterUserCatalogController.class.getName());
+
+    public WaterUserCatalogController(MetricRegistry metrics) {
+        waterMetrics(metrics);
+    }
+
+    @OpenApi(
+        pathParams = {
+            @OpenApiParam(name = OFFICE, description = "The office Id the water user is associated with.",
+                    required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "The project Id the water user is associated with.",
+                    required = true)
+        },
+        responses = {
+            @OpenApiResponse(status = STATUS_200,
+                content = {
+                    @OpenApiContent(type = Formats.JSONV1, from = WaterUser.class)
+                })
+        },
+        description = "Gets all water users.",
+        method = HttpMethod.GET,
+        path = "/projects/{office}/{project-id}/water-user/all-users",
+        tags = {TAG}
+    )
+
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(GET_ALL)) {
+            DSLContext dsl = getDslContext(ctx);
+            String office = ctx.pathParam(OFFICE);
+            String locationId = ctx.pathParam(PROJECT_ID);
+            CwmsId projectLocation = buildCwmsId(office, locationId);
+            String result;
+            String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) : Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, WaterUserContract.class);
+            ctx.contentType(contentType.toString());
+            WaterContractDao contractDao = getContractDao(dsl);
+            List<WaterUser> users = contractDao.getAllWaterUsers(projectLocation);
+
+            if (users.isEmpty()) {
```

Comment:
don't throw an error if the list is empty

Reply:
Removed unnecessary error handling